PeerioTechnologies/peerio-desktop
# chore: React console errors on prod only

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7506
#### Linked PRs in other repos
n/a
#### Manual testing instructions  
Quick test by me, not yet by @ebilg 

Some of these changes are pretty broad, for example pretty much the entire app is wrapped in a new `<div>` in both App.jsx and Root.jsx, so there is a small potential for layout to break, well, pretty much anywhere. (It will be obvious if so). Because I needed to look through every view to check for console errors anyway, I should have caught these cases, but I may have missed something.

In order to test this you need to go into `package.json` and add `NODE_ENV=production` to the "start" script (first line below "MAIN COMMANDS"), so the full line reads `"start": "NODE_ENV=production npm run compile && npm-run-all -l -p code:watch icebear:watch style:watch devstyle:watch electron:run"`

#### Additional notes
n/a

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
